Public/Src/Engine/Scheduler/Distribution/ExecutionResultSerializer.cs

```diff
@@ -613,5 +621,51 @@ private void WriteFileArtifact(BuildXLWriter writer, FileArtifact file)
             WriteAbsolutePath(writer, file.Path);
             writer.WriteCompact(file.RewriteCount);
         }
+
+        private static void ReadPipProperties(BuildXLReader reader, out IReadOnlyDictionary<string, int> pipProperties)
+        {
+            bool hasPipProperties = reader.ReadBoolean();
+
+            if (!hasPipProperties)
+            {
+                pipProperties = null;
+            }
+            else
+            {
+                int count = reader.ReadInt32Compact();
+
+                var tmpPipProperties = new Dictionary<string, int>();

```

Comment:
Why do we need a temporary variable? Can the method just return the dictionary instead of using `out`?

Reply:
I was trying to match the other methods in this file used by the serializer, though I agree just returning is clearer.  Updated.